aspnetcore/signalr/javascript-client.md

```diff
@@ -98,7 +98,140 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
+
+```javascript
+connection.onreconnecting((error) => {
+  console.assert(connection.sate === signalR.HubConnectionState.Reconnecting);
+
+  document.getElementById("messageInput").disabled = true;
+
+  const li = document.createElement("li");
+  li.textContent = `Connection lost due to error "${error}". Reconnecting.`;
+  document.getElementById("messagesList").appendChild(li);
+});
+```
+
+If the client successfully reconnect within its first four attempts, the `HubConnection` will transition back to the [Connected](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onreconnected](/javascript/api/%40aspnet/signalr/hubconnection#onreconnected) callbacks. This is a good opportunity to inform users the connection has been reestablished.
+
+Since the connection looks entirely new to the server, a new connectionId will be provided to the onreconnected callback.
+
+> [!WARNING]
+> The onreconnected callbacks connectionId parameter will not be undefined if the `HubConnection` was configured to [skip negotiation.](xref:signalr/configuration#configure-client-options)
+
+```javascript
+connection.onreconnected((connectionId) => {
+  console.assert(connection.sate === signalR.HubConnectionState.Connected);
+
+  document.getElementById("messageInput").disabled = false;
+
+  const li = document.createElement("li");
+  li.textContent = `Connection reestablished. Connected with connectionId "${connectionId}".`;
+  document.getElementById("messagesList").appendChild(li);
+});
+```
+
+`withAutomaticReconnect()` will not configure the HubConnection to retry initial start failures, so start failures need to be handled manually:
+
+```javascript
+async function start() {
+    try {
+        await connection.start();
+        console.assert(connection.sate === signalR.HubConnectionState.Connected);
+        console.log("connected");
+    } catch (err) {
+        console.assert(connection.sate === signalR.HubConnectionState.Disconnected);
+        console.log(err);
+        setTimeout(() => start(), 5000);
+    }
+};
+```
+
+If the client doesn't succesfully reconnect within its first four attempts, the `HubConnection` will transition to the [Disconnected](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onclosed](/javascript/api/%40aspnet/signalr/hubconnection#onreconnected) callbacks. This is a good opportunity to inform users the connection has been permanently lost and recommend refreshing the page:
```

Comment:
> This is a good opportunity to inform users the connection has been permanently lost and recommend refreshing the page

Epic. Love how you provide the first example as a place to disable UI and this as a place to pull the rip cord. Great guidance in the docs, @halter73. Much happy.